docs/config.rst

```diff
@@ -353,13 +348,83 @@ Or, if the option needs to be available as a function parameter::
 
 
-See Also
-========
+.. _config-affiliated:
+
+Affiliated Packages
+===================
+
+If you maintain an affiliated package you can customise the behaviour of the
+astropy configuration machinery for your package. Prior to version 4.0 of
+Astropy the bahviour of both ``astropy`` and the package template was to write a
+configuration file to ``~/.astropy/yourpackage.cfg`` if a template configuration
+file was provided by your package. If you wish to maintain this behaviour you do
+not have to change anything.
+
+If you want to change the behaviour of your package to mimic that of ``astropy``
+you can define a wrapper for `astropy.config.write_default_config` in your
+``__init__.py`` file i.e.::
+
+
+  def write_default_config():
+      """
+      Writes out the template configuration file for this version of packagename.
+
+      This function will save a template config file for manual editing, if a
+      config file already exists, this will write a config file appended with the
+      version number, to facilitate comparison of changes.
+
+      Returns
+      -------
+      filepath : `pathlib.Path` or `None`
+          The full path of the file written or `None` if no file was written.
+      """
+      from . import config as _config
+      return _config.write_default_config('packagename')
+
+
+This function provides a way for your users to write out a version of your
+configuration template, in the same way that they are written on import before
+this change.
+
 
-.. toctree::
-   :maxdepth: 2
+Customizing the Configuration Directory
+---------------------------------------
 
-   config_0_4_transition
+.. warning::
+
+   If you used the :ref:`package template <packagetemplate:package-template>` to
+   create your affiliated package this might cause you to have to be more
+   careful when pulling in updates from the package template until the template
+   is updated to reflect these changes.
+
+
+Starting with the 4.0 release of astropy it is possible to override the name of
+the directory where the configuration files are stored. Before 4.0 and by
+default this is the ``.astropy`` directory (often ``~/.astropy``). To change
+this you can set the ``rootname=`` keyword in various places. To do this you
+need to make two changes, firstly, you need to add a ``rootname='packagename'``
+argument to the ``_config.write_default_config`` call in the previous example.
+Secondly you need to override the `astropy.config.ConfigItem` class. The best
+way to do that is to define a ``config`` submodule for your package i.e.
+``packagename/config/__init__.py`` would look like this::
+
+  
+  from astropy.config import ConfigNamespace, ConfigItem as _AstropyConfigItem
+
+
+  __all__ = ['ConfigItem', 'ConfigNamespace']
+
+
+  class ConfigItem(_AstropyConfigItem):
+      rootname = 'packagename'
+
+
+You would then import ``ConfigNamespace`` and ``ConfigItem`` from
+``packagename.config``.
```

Comment:
Also, maybe add a note that it is the affiliated package maintainer's responsibility to communicate this customization and that it means for the package's backward compatibility to their users. (So their users won't come here to complain because we can't do anything about it anyway.)